newIDE/app/src/MainFrame/index.js

```diff
@@ -1188,6 +1194,35 @@ class MainFrame extends React.Component<Props, State> {
     }
   };
 
+  saveAs = () => {
+    this.setState({
+      isSaveAs: true,
+    });
+    saveUiSettings(this.state.editorTabs);
+
+    const { currentProject, isSaveAs } = this.state;
+    if (!currentProject) return;
+    const { i18n } = this.props;
+
+    if (this.props.saveDialog) {
+      this._openSaveDialog();
+    } else if (this.props.onSaveProject) {
+      this.props.onSaveProject(currentProject, isSaveAs).then(
```

Comment:
~~Prefer passing true here (or even better, an object: `{ isSaveAs: true }`, so that it's clearer)~~

In fact, prefer calling this.props.onSaveProjectAs